src/flamegraphs.jl

```diff
@@ -0,0 +1,205 @@
+    # its children. That is, if the exclusive time of any node is non-zero.
+    # AND, we want to enter each "section" of exclusive time as a separate sample, in case
+    # that allows PProf to do any ordering in its display.
+    # So, the algorithm is that we enter a sample for every gap between the spans of the
+    # children.
+    function emit_tree(node)
+        span = node.data.span
+        start = span.start
+        stop = span.stop
+        child = node.child
+        if child === node
+            # We're a leaf node, so just emit this node directly, and then we're done.
+            emit_stack_sample(child, start:stop)
+            return
+        end
+        while true  # break when we've exhausted the children
+            cspan = child.data.span
+            if cspan.start > start
+                # We know there's a gap before the next child starts
+                # So we emit a frame for `node`
+                emit_stack_sample(node, start:cspan.start-1)
+            end
+            # Now emit the child node
+            emit_tree(child)
+
+            start = cspan.stop + 1
+
+            if child.sibling === child break end
+            child = child.sibling
+        end
+        # If there was still time after the last child, that's another gap that we can emit.
+        cspan = child.data.span
+        if stop > cspan.stop
+            # We know there's a gap before the next child starts
+            # So we emit a frame for `node`
+            emit_stack_sample(node, start:stop)
+        end
+    end
+
+    function emit_stack_sample(leaf, span)
+        location_id = Vector{Any}()
+
+        # Walk back up the callstack, collecting all stack traces along the way, building
+        # up a stack trace for this span.
+        node = leaf
+        while true  # do-while node.parent === node    -- this makes sure we get the top node.
+            data = node.data
+
+            if !from_c && data.sf.from_c
+                node = node.parent
+                continue
+            end
+
+            frame = data.sf
+            # Don't use the ip pointer, because this data is provided by the user and isn't
+            # guaranteed to be unique.
+            id = hash(frame)
+
+            location = Location(;id = id, address = frame.pointer, line=[])
+            push!(location_id, id)
+            locs[id] = location
+            linfo = data.sf.linfo
+
+            func_id = if frame.linfo !== nothing
+                hash(frame.linfo)
+            else
+                hash((frame.func, frame.file, frame.line))
+            end
+
+            _register_function(funcs, func_id, linfo, frame)
+            push!(location.line, Line(function_id = func_id, line = frame.line))
+
+            if node.parent === node
+                break
+            end
+
+            node = node.parent
+        end
+
+        value = [
+            length(span), # time duration (nanoseconds?)
```

Comment:
You switched to from `nanoseconds` to counts. So should this be `length(span)/period`?

Reply:
Oops, no, i just need to fix the comment.

I was actually just totally screwed up before. I _though_ the span was nanoseconds, but it turns out the span is already sample counts, which is why I switched. 

I think in general it might be nice to have a setting in PProf that uses the `period` to display time instead, though, yeah!

Reply:
```suggestion
            length(span), # Number of samples in this frame.
```

Reply:
(done. lemme know if this looks right now? Thanks)